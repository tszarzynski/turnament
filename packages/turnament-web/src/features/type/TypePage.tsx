import { Button, Hr, IconNext } from "turnament-components";
import type { SchedulerType } from "turnament-scheduler";
import { routes } from "../../app/router";
import { useBaseStore } from "../../app/store";
import PageHeader from "../../components/PageHeader";
import { schedulerOptions } from "../../utils/schedulerUtils";
import PageLayout, { PageContent, PageBody } from "../../components/PageLayout";
import PageNavigation from "../../components/PageNavigation";

const TypePage = () => {
	const nextRound = useBaseStore((state) => state.nextRound);
	const schedulerType = useBaseStore((state) => state.schedulerType);
	const setSchedulerType = useBaseStore((state) => state.setSchedulerType);

	const handleOptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		setSchedulerType(event.target.value as SchedulerType);
	};

	const handleNext = () => {
		nextRound();
		routes.tournamentRounds().push();
	};

	return (
		<PageLayout>
			<PageContent>
				<PageHeader>Choose Turnament Type</PageHeader>
				<PageBody>
					<fieldset>
						<legend className="sr-only">Tournament Type</legend>
						<div className="border-2 border-secondary p-0.5">
							<Hr />
							{schedulerOptions.map((option) => (
								<div
									key={option.value}
									className="mb-0.5 flex w-full flex-row items-stretch justify-between border border-secondary"
								>
									<label
										htmlFor={option.value}
										className="flex flex-1 items-center border-secondary border-r px-4 py-2 font-medium text-lg text-secondary uppercase"
									>
										{option.name}
									</label>
									<input
										id={option.value}
										name="tournament-type"
										type="radio"
										value={option.value}
										checked={schedulerType === option.value}
										onChange={handleOptionChange}
										className="h-10 w-10 bg-primary accent-black focus:outline-none focus:ring-2 focus:ring-black focus:ring-offset-1"
									/>
								</div>
							))}
						</div>
					</fieldset>
				</PageBody>
			</PageContent>
			<PageNavigation>
				<Button
					type="button"
					fullWidth={false}
					iconSlot={<IconNext />}
					onClick={handleNext}
				>
					Start
				</Button>
			</PageNavigation>
		</PageLayout>
	);
};

export default TypePage;
