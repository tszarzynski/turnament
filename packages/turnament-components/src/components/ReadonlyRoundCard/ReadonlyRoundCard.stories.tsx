import type { Match } from "turnament-scheduler";
import ReadonlyRoundCard from "./ReadonlyRoundCard";

export default {
	component: ReadonlyRoundCard,
	title: "ReadonlyRoundCard",
};

const match: Match = {
	ID: "ID",
	roundID: 1,
	hasBye: false,
	pairing: [1, 2],
	result: [0, 0],
};

const players = [
	{ name: "Name1", active: true, ID: 1 },
	{ name: "Name2", active: true, ID: 2 },
];

export const Default = () => {
	return (
		<ReadonlyRoundCard
			matches={[match, match, match]}
			roundNum={1}
			players={players}
		/>
	);
};
