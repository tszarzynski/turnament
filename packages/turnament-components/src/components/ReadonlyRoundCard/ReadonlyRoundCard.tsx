import type { Match, Player } from "turnament-scheduler";
import MatchCard from "../MatchCard";
import usePlayerNames from "../usePlayerNames";

type Props = {
	matches: Match[];
	players: Player[];
	roundNum: number;
};

const ReadonlyRoundCard = ({ matches, players }: Props) => {
	const names = usePlayerNames(players, matches);

	return (
		<div className="flex flex-col gap-0.5 border-2 border-gray-300 p-0.5">
			{matches.map((match, i) => (
				<MatchCard
					key={match.ID}
					match={match}
					names={names[i]}
					disabled={true}
				/>
			))}
		</div>
	);
};

export default ReadonlyRoundCard;
