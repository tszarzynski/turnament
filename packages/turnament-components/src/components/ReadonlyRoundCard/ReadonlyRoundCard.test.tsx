import { render } from "vitest-browser-react";
import ReadonlyRoundCard from "./ReadonlyRoundCard";
import type { Match } from "turnament-scheduler";

const match: Match = {
	ID: "ID",
	roundID: 1,
	hasBye: false,
	pairing: [1, 2],
	result: [0, 0],
};

const players = [
	{ name: "Name1", active: true, ID: 1 },
	{ name: "Name2", active: true, ID: 2 },
];

describe("ReadonlyRoundCard", () => {
	it("renders without error", () => {
		const { asFragment } = render(
			<ReadonlyRoundCard
				matches={[match, match, match]}
				roundNum={1}
				players={players}
			/>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
