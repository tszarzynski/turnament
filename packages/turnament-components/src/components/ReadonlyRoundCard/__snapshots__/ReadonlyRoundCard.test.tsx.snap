// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReadonlyRoundCard > renders without error 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-0.5 border-2 border-gray-300 p-0.5"
  >
    <div
      class="flex flex-col"
    >
      <div
        class="focus-within:z-10"
      >
        <div
          class="flex flex-row items-stretch justify-between gap-0.5"
        >
          <h5
            class="flex flex-auto items-center border pl-3 font-bold text-lg border-gray-300 text-gray-300"
          >
            Name1
          </h5>
          <div
            class="flex flex-row "
          >
            <form
              autocomplete="off"
            >
              <input
                autocomplete="off"
                autocorrect="off"
                class="focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent h-[54px] w-[54px] text-center"
                disabled=""
                name="score"
                readonly=""
                type="text"
                value="0"
              />
            </form>
          </div>
        </div>
      </div>
      <div
        class="-mt-[1px] focus-within:z-10"
      >
        <div
          class="flex flex-row items-stretch justify-between gap-0.5"
        >
          <h5
            class="flex flex-auto items-center border pl-3 font-bold text-lg border-gray-300 text-gray-300"
          >
            Name2
          </h5>
          <div
            class="flex flex-row "
          >
            <form
              autocomplete="off"
            >
              <input
                autocomplete="off"
                autocorrect="off"
                class="focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent h-[54px] w-[54px] text-center"
                disabled=""
                name="score"
                readonly=""
                type="text"
                value="0"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <div
        class="focus-within:z-10"
      >
        <div
          class="flex flex-row items-stretch justify-between gap-0.5"
        >
          <h5
            class="flex flex-auto items-center border pl-3 font-bold text-lg border-gray-300 text-gray-300"
          >
            Name1
          </h5>
          <div
            class="flex flex-row "
          >
            <form
              autocomplete="off"
            >
              <input
                autocomplete="off"
                autocorrect="off"
                class="focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent h-[54px] w-[54px] text-center"
                disabled=""
                name="score"
                readonly=""
                type="text"
                value="0"
              />
            </form>
          </div>
        </div>
      </div>
      <div
        class="-mt-[1px] focus-within:z-10"
      >
        <div
          class="flex flex-row items-stretch justify-between gap-0.5"
        >
          <h5
            class="flex flex-auto items-center border pl-3 font-bold text-lg border-gray-300 text-gray-300"
          >
            Name2
          </h5>
          <div
            class="flex flex-row "
          >
            <form
              autocomplete="off"
            >
              <input
                autocomplete="off"
                autocorrect="off"
                class="focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent h-[54px] w-[54px] text-center"
                disabled=""
                name="score"
                readonly=""
                type="text"
                value="0"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <div
        class="focus-within:z-10"
      >
        <div
          class="flex flex-row items-stretch justify-between gap-0.5"
        >
          <h5
            class="flex flex-auto items-center border pl-3 font-bold text-lg border-gray-300 text-gray-300"
          >
            Name1
          </h5>
          <div
            class="flex flex-row "
          >
            <form
              autocomplete="off"
            >
              <input
                autocomplete="off"
                autocorrect="off"
                class="focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent h-[54px] w-[54px] text-center"
                disabled=""
                name="score"
                readonly=""
                type="text"
                value="0"
              />
            </form>
          </div>
        </div>
      </div>
      <div
        class="-mt-[1px] focus-within:z-10"
      >
        <div
          class="flex flex-row items-stretch justify-between gap-0.5"
        >
          <h5
            class="flex flex-auto items-center border pl-3 font-bold text-lg border-gray-300 text-gray-300"
          >
            Name2
          </h5>
          <div
            class="flex flex-row "
          >
            <form
              autocomplete="off"
            >
              <input
                autocomplete="off"
                autocorrect="off"
                class="focus:none border-1 border-black font-bold text-lg placeholder:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-1 border-gray-300 text-gray-300 cursor-not-allowed hover:bg-transparent h-[54px] w-[54px] text-center"
                disabled=""
                name="score"
                readonly=""
                type="text"
                value="0"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
